aws/sts.go

```diff
@@ -2,25 +2,46 @@ package aws
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/fatih/color"
 )
 
-func AssumeSAMLRole(PrincipalArn, RoleArn, SAMLAssertion string) (*Credentials, error) {
+// AssumeSAMLRole asumes a Role using the SAMLAssertion specified. If the duration cannot be meet it transperently lowers the duration and sets the returned bool to true to signal that a message should be displayed
+func AssumeSAMLRole(PrincipalArn, RoleArn, SAMLAssertion, profile string, duration int64) (*Credentials, error) {
+	creds, err, roleNeedsChange := assumeSAMLRole(PrincipalArn, RoleArn, SAMLAssertion, duration, false)
+	if err == nil && roleNeedsChange {
+		// TODO: This might conflict with spinner... Trying to avoid by starting with \r.
```

Comment:
The problem with the lower call stack is that we'll end up with: 
a) duplicate code and
b) more logic.

The idea is that the duration of the credentials is lower in 1h steps, because 
a) not all roles will support extended session support, 
b) this can be changed any time on the AWS account and then bailing out isn't the nicest solution. 

So it trys to find the highest supported timeout that is lower than what the user requested.